engine/src/main/java/io/camunda/zeebe/engine/processing/timer/TriggerTimerProcessor.java

```diff
@@ -171,4 +179,19 @@ private void rescheduleTimer(
         writer,
         sideEffects::accept);
   }
+
+  private Interval refreshInterval(final long lastDueDate, final Interval interval) {
+    if (interval.getStart() == null) {
+      return interval;
+    }
+
+    final Instant start = Instant.ofEpochMilli(lastDueDate);
+    final ZoneId zoneId =
+        Optional.ofNullable(interval.getStart())
+            .map(ZonedDateTime::getZone)
+            .orElse(ZoneId.systemDefault());
```

Comment:
:wrench: We could remove the `Optional` usage here. We do a `null` check in the method already.